web/src/state_data.ts

```diff
@@ -0,0 +1,87 @@
+import type {GroupPermissionSetting} from "./types";
+
+export let state_data: {
+    avatar_source: string;
+    custom_profile_fields: {
+        display_in_profile_summary?: boolean;
+        field_data: string;
+        hint: string;
+        id: number;
+        name: string;
+        order: number;
+        type: number;
+    }[];
+    delivery_email: string;
+    is_admin: boolean;
+    is_billing_admin: boolean;
+    is_guest: boolean;
+    is_moderator: boolean;
+    is_owner: boolean;
+    max_avatar_file_size_mib: number;
+    max_icon_file_size_mib: number;
+    max_logo_file_size_mib: number;
+    max_message_id: number;
+    muted_users: {id: number; timestamp: number}[];
+    realm_add_custom_emoji_policy: number;
+    realm_available_video_chat_providers: {
+        disabled: {name: string; id: number};
+        jitsi_meet: {name: string; id: number};
+        zoom?: {name: string; id: number};
+        big_blue_button?: {name: string; id: number};
+    };
+    realm_avatar_changes_disabled: boolean;
+    realm_bot_domain: string;
+    realm_can_access_all_users_group: number;
+    realm_create_multiuse_invite_group: number;
+    realm_create_private_stream_policy: number;
+    realm_create_public_stream_policy: number;
+    realm_create_web_public_stream_policy: number;
+    realm_delete_own_message_policy: number;
+    realm_description: string;
+    realm_edit_topic_policy: number;
+    realm_email_changes_disabled: boolean;
+    realm_enable_guest_user_indicator: boolean;
+    realm_enable_spectator_access: boolean;
+    realm_icon_source: string;
+    realm_icon_url: string;
+    realm_invite_to_realm_policy: number;
+    realm_invite_to_stream_policy: number;
+    realm_is_zephyr_mirror_realm: boolean;
+    realm_jitsi_server_url: string | null;
+    realm_logo_source: string;
+    realm_logo_url: string;
+    realm_move_messages_between_streams_policy: number;
+    realm_name_changes_disabled: boolean;
+    realm_name: string;
+    realm_night_logo_source: string;
+    realm_night_logo_url: string;
+    realm_notifications_stream_id: number;
+    realm_org_type: number;
+    realm_plan_type: number;
+    realm_private_message_policy: number;
+    realm_push_notifications_enabled: boolean;
+    realm_upload_quota_mib: number | null;
+    realm_uri: string;
+    realm_user_group_edit_policy: number;
+    realm_video_chat_provider: number;
+    realm_waiting_period_threshold: number;
+    server_avatar_changes_disabled: boolean;
+    server_jitsi_server_url: string | null;
+    server_name_changes_disabled: boolean;
+    server_needs_upgrade: boolean;
+    server_presence_offline_threshold_seconds: number;
+    server_supported_permission_settings: {
+        realm: Record<string, GroupPermissionSetting>;
+        stream: Record<string, GroupPermissionSetting>;
+        group: Record<string, GroupPermissionSetting>;
+    };
+    server_web_public_streams_enabled: boolean;
```

Comment:
I think I'd like to have the variables with `server_` or `realm_` prefixes go into a couple modules, `realm_settings.ts` and `server_settings.ts`, with implementations parallel to `web/src/user_settings.ts`. We have a concept in the API design that we'll make separate `realm` and `server` objects in the API, and doing so would achieve most of the web app refactoring required to support that. Presumably that would mean you'd access `realm_settings.private_message_policy`, eliminating the need for conversions to convert between whether there's a `realm_` prefix on a field outside the new `realm_settings.ts` module.

Probably a bunch of the `*_size_mb` constants as well as `zulip_version` type fields would go into `server_settings` too. 

(I think the remaining fields, like `page_params.full_name` can go into a `state_data` object, for lack of a more clearly defined future home).

Reply:
I’m not sure there’s a bright line along which such further splitting can happen or a clear case for doing that, but I’ll start a [discussion](https://chat.zulip.org/#narrow/stream/6-frontend/topic/Splitting.20page_params/near/1713115).